File: javascript/basico/examples/bucle_while__while.js
/**
 * Objetivo: repetir código mientras se cumpla una condición
 * Referencia: while
 * Tipo: keyword
 * Nivel: básico
 */

// While básico
let contador = 0;

while (contador < 3) {
    console.log("Contador: " + contador);
    contador++;
}

// While con condición variable
let creditos = 5;

while (creditos > 0) {
    console.log("Créditos restantes: " + creditos);
    creditos--;
}

// While con lógica
let entrada = "";
let intentos = 0;

// Simulación (en el navegador pedirías input)
let intentosMaximos = 2;
while (entrada !== "contraseña" && intentos < intentosMaximos) {
    entrada = "intento"; // Cambiar para evitar bucle infinito
    intentos++;
    console.log("Intento " + intentos);
}

/*
output
Contador: 0
Contador: 1
Contador: 2
Créditos restantes: 5
Créditos restantes: 4
Créditos restantes: 3
Créditos restantes: 2
Créditos restantes: 1
Intento 1
Intento 2
*/

File: python/basico/examples/bucle_while__while.py
"""
Objetivo: Iterar mientras una condición sea verdadera
Referencia: while
Tipo: keyword
Nivel: basico
"""

# while simple
contador = 0
while contador < 3:
    print(contador)
    contador += 1

print("---")

# while con variable booleana
activo = True
numero = 5
while activo and numero > 0:
    print(numero)
    numero -= 1
    if numero == 0:
        activo = False

"""output
0
1
2
---
5
4
3
2
1
"""
